agent/structs/connect_proxy_config_test.go

```diff
@@ -179,6 +181,22 @@ func TestUpstream_UnmarshalJSON(t *testing.T) {
 			},
 			wantErr: false,
```

Comment:
What if it returned an error? Otherwise someone could think the field was accepted

Reply:
It does return an error, this is for the test case above. 

https://github.com/hashicorp/consul/pull/7678/files/e0c3660e3fd6fa30454c1cccde9afe2f214503ce#diff-5075783e5f891bd7f3d2f1d5769f57acR198

Reply:
Or, if you were talking about the `TestUpstream_MarshalJSON` case, that is testing the conversion from `structs.Upstream` -> `JSON`, and thus `IngressHosts` can only be set by code inside the code base.